src/modal/modal.js

```diff
@@ -265,22 +265,22 @@ angular.module('ui.bootstrap.modal', [])
 
       function removeModalWindow(modalInstance, elementToReceiveFocus) {
 
-        var body = $document.find('body').eq(0);
         var modalWindow = openedWindows.get(modalInstance).value;
+        var modalContainer = angular.element($document[0].querySelector(modalWindow.appendTo)).eq(0);
```

Comment:
What if you are launching the modal inside a directive and pass the angular `element` from the directive `link` function? or `$element` from the directive controller? Thats what i'm looking for right now :D

like:

```javascript
.directive('someDirective', [function () {
	return {
		restrict: 'C',
		template: '<button ng-click="clickMe()">blabla</button',
		scope: true,
		replace: true,
		controller: ['$scope', '$modal', '$element', function ($scope, $modal, $element) {

			$scope.clickMe = function () {
				$modal.open({
					templateUrl: 'someUrl',
					appendTo: $element, //$element.parent() ....
					controller: ['$scope', '$modalInstance', function ($scope, $modalInstance) {

						$scope.close = function() {
							$modalInstance.dismiss('cancel');
						}

					}]
				})
			};

		}]
	};
}])
```